pkg/kapp/cmd/app/deploy.go

```diff
@@ -448,6 +457,72 @@ func (o *DeployOptions) calculateAndPresentChanges(existingResources,
 	return clusterChangeSet, clusterChangesGraph, (len(clusterChanges) == 0), changesSummary, err
 }
 
+func CheckForPriodicAnnotation(existingResources, newResources []ctlres.Resource) error {
+
+	var (
+		duration time.Duration
+		err      error
+	)
+
+	addNonceMod := ctlres.StringMapAppendMod{
+		ResourceMatcher: ctlres.AllMatcher{},
+		Path:            ctlres.NewPathFromStrings([]string{"metadata", "annotations"}),
+		KVs: map[string]string{
+			lastRenewTime: fmt.Sprintf("%v", time.Now().UTC().Format(time.RFC3339)),
+		},
+	}
+	existRes := existingPriodicResource(existingResources)
+
+	for _, res := range newResources {
+		rs, found := existRes[res.Name()+res.Kind()]
+		if found {
+			val, found := rs.Annotations()[maxDuration]
+			if !found {
```

Comment:
What happens if the the max-duration annotation is added to an already existing resource for which the max-duration has passed since it was created? 
What happens if the annotation is removed for an existing resource with the annotation present?

Reply:
- If we will add max-duration to an existing resource and condition (creationTimestamp + max-duration < current time) has passed then for versioned resource, new resource will get created while for non-versioned resource, resource will get updated with updated strategy always-replace.
-  If we will remove max-duration from an existing resource then normal update will happen. 

Added e2e test case for the same.